pkg/common/client.go

```diff
@@ -19,172 +18,117 @@ import (
-		compute.NewDisksClientWithBaseURI(config.EnvironmentEndpoint, config.SubscriptionID),
-		compute.NewVirtualMachineScaleSetsClientWithBaseURI(config.EnvironmentEndpoint, config.SubscriptionID),
-		network.NewInterfacesClientWithBaseURI(config.EnvironmentEndpoint, config.SubscriptionID),
+func NewClient(configMode string) (client AzureClient) {
+	var configload Config
+	if configMode == "metadata" {
+		configload = LoadConfig()
+	} else if configMode == "environment" {
+		configload = EnvLoadConfig()
+	} else {
+		log.Print("Invalid config Mode")
 	}
 
-	// Authorizing with Managed Service Identity
-	err := os.Setenv("AZURE_ENVIRONMENT", config.AzureEnvironment)
-	if err != nil {
-		log.Panicf("Unable to set Azure environment: %v", err)
-	}
-	authorizer, err := auth.NewAuthorizerFromEnvironment()
-	if err == nil {
-		client.VirtualMachinesClient.Authorizer = authorizer
-		client.DisksClient.Authorizer = authorizer
-		client.VirtualMachineScaleSetsClient.Authorizer = authorizer
-		client.InterfacesClient.Authorizer = authorizer
+	client = AzureClient{
+		configload,
+		GetVMClient(configload),
+		GetNicClient(configload),
+		GetLbClient(configload),
 	}
-
-	client.VirtualMachinesClient.RetryAttempts = 1
-	client.DisksClient.RetryAttempts = 1
-	client.VirtualMachineScaleSetsClient.RetryAttempts = 1
-	client.InterfacesClient.RetryAttempts = 1
-
 	return
 }
 
-// GetVM Returns a VirtualMachine object.
-func (c AzureClient) GetVM(vmname string) (vm compute.VirtualMachine) {
-	ctx, cancel := context.WithTimeout(context.Background(), 6000*time.Second)
-	defer cancel()
-
-	vm, err := c.VirtualMachinesClient.Get(ctx, Conf.ResourceGroup, vmname, compute.InstanceView)
+//GetVMClient return vmClient
+func GetVMClient(configload Config) compute.VirtualMachinesClient {
+	vmClient := compute.NewVirtualMachinesClient(configload.SubscriptionID)
+	a, err := auth.NewAuthorizerFromEnvironment()
 	if err != nil {
-		log.Panicf("failed to get VM: %v", err)
+		glog.Fatalf("failed to create authorizer from environment: %s\n", err)
 	}
-
-	return
+	vmClient.Authorizer = a
+	vmClient.AddToUserAgent("sdk-samples")
+	return vmClient
 }
 
-// GetAllVM Returns a ListResultPage of all VMs in the ResourceGroup of the Config
-func (c AzureClient) GetAllVM() (result compute.VirtualMachineListResultPage) {
-	ctx, cancel := context.WithTimeout(context.Background(), 6000*time.Second)
-	defer cancel()
-
-	result, err := c.VirtualMachinesClient.List(ctx, Conf.ResourceGroup)
+// GetNicClient return nic client
+func GetNicClient(configload Config) network.InterfacesClient {
+	nicClient := network.NewInterfacesClient(configload.SubscriptionID)
+	a, err := auth.NewAuthorizerFromEnvironment()
 	if err != nil {
-		log.Panicf("failed to get all VMs: %v", err)
+		glog.Fatalf("failed to create authorizer from environment: %s\n", err)
 	}
-
-	return
+	nicClient.Authorizer = a
+	nicClient.AddToUserAgent("sdk-samples")
```

Comment:
Can we use "azure-request-limitometer" as user agent please?

Reply:
Updated with azure-request-limitometer-Nic